### PR TITLE
Convert date string to date object in forms

### PR DESCRIPTION
**Pull request by tazo90**
From Angular 1.3+ all date-related inputs like <input type="date"> require the model to be a Date object.

### CODE REVIEW

**Comment by reupen** [comment]
Hi @tazo90,

Good work! A few small things we can improve upon:

*  Can we fix datetime fields not being populated as well?
*  I think we want to check the type of modelValue in the formatter function. If it's a string or a number, then pass it to the Date constructor, otherwise just return modelValue.

**Comment by tazo90** [comment]
I've added fixes.

File: luxjs/forms/form.js [hunk 1]
```diff
@@ -309,6 +314,10 @@
                     // Add model attribute
                     input.attr('ng-model', scope.formModelName + '["' + field.name + '"]');
 
+                    // Add default placeholder to date field
+                    if (field.type === 'date')
+                        field.placeholder = 'YYYY-MM-DD';
```
Comment (lsbardel) [inline luxjs/forms/form.js]: You are overriding the placeholder here, what about if it was already available?

File: luxjs/forms/form.js [hunk 2]
```diff
@@ -223,6 +223,11 @@
                     // lux-codemirror directive
                     if (scope.field.hasOwnProperty('text_edit'))
                         element.attr('lux-codemirror', scope.field.text_edit);
+
+                    // Convert date string to date object
+                    var dateTypes = ['date', 'datetime', 'datetime-local'];
```
Comment (lsbardel) [inline luxjs/forms/form.js]: can this be specified somewhere else, maybe as a constant?

**Comment by lsbardel** [comment]
We need to add tests

**Comment by tazo90** [comment]
@lsbardel I did updates, please review in free time. I hope it will be fine.

**Comment by lsbardel** [comment]
@tazo90 I can't review this!!
can you please put your commits only?

**Comment by tazo90** [comment]
@lsbardel this PR can be close, I created another one PR for this issue: [https://github.com/quantmind/lux/pull/194](https://github.com/quantmind/lux/pull/194)

**Pull request closed without merging**